src/formats/onenote.ts

```diff
@@ -142,150 +147,174 @@ export class OneNoteImporter extends FormatImporter {
 
 		progress.status('Starting OneNote import');
 
-		for (let section of this.selectedSections) {
+		for (let sectionId of this.selectedIds) {
+			this.currentSection = sectionId;
 			progress.reportProgress(0, remainingSections);
 			remainingSections--;
-
 			let pageCount: number = 0;
 
-			let sectionFolder: TFolder;
-			if (section.parentSectionGroup) {
-				let sectionGroupFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.parentSectionGroup.displayName);
-				sectionFolder = await this.createFolders(sectionGroupFolder.path + '/' + section.displayName);
-			}
-			else sectionFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
+			const baseUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${sectionId}/pages`;
+			const params = new URLSearchParams({
+			  $select: 'id,title,createdDateTime,lastModifiedDateTime,level,order,contentUrl',
+			  $orderby: 'order',
+			  pagelevel: 'true'
+			});
 
-			const pagesUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`;
-			let pages: OnenotePage[] = (await this.fetchResource(pagesUrl, 'json')).value;
+			const pagesUrl = `${baseUrl}?${params.toString()}`;
+	
+			let pages: OnenotePage[] = ((await this.fetchResource(pagesUrl, 'json')).value);
+			this.currentPages = pages;
 
 			progress.reportProgress(0, pages.length);
 
 			for (let i = 0; i < pages.length; i++) {
 				const page = pages[i];
-
+				if (!page.title) page.title = `Untitled-${moment().format('YYYYMMDDHHmmss')}`;
 				try {
 					pageCount++;
-					progress.status(`Importing note ${page.title || 'Untitled'}`);
+					progress.status(`Importing note ${page.title}`);
 
 					// Every 50 items, do a few second break to prevent rate limiting
 					if (i !== 0 && i % 50 === 0) {
-						await new Promise(resolve => setTimeout(resolve, 5000));
+						await new Promise(resolve => setTimeout(resolve, 7500));
 					}
-					this.processFile(progress,
-						sectionFolder,
-						outputFolder,
-						await this.fetchResource(`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`, 'text')
-						, page);
 
+					this.processFile(progress,
+						await this.fetchResource(`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`, 'text'),
+						page);
 					progress.reportProgress(pageCount, pages.length);
 
 				}
 				catch (e) {
-					progress.reportFailed(page.title || 'Untitled note', e.toString());
+					progress.reportFailed(page.title, e.toString());
 				}
 			}
 		}
 	}
 
-	async processFile(progress: ImportContext, sectionFolder: TFolder, outputFolder: TFolder, content: string, page: OnenotePage) {
+	async processFile(progress: ImportContext, content: string, page: OnenotePage) {
 		try {
+			// Removes multiple paragraphs in a row to replace them with a basic <br>
+			const paragraphRegex: RegExp = /<\/p>\s*<p[^>]*>/g;
```

Comment:
These constants can be declared outside of this function. At the top of the file would probably be a good place for them.

Reply:
Hmm, but if those are only used in this one function, are you sure they should be at the top of the file?

Reply:
I mainly meant it would be good to pull them out of this function since they are being re-instantiated every time the function is called.

Reply:
Thanks for noticing, that's a neat optimization.

Reply:
Looks like these still need to be moved.